### PR TITLE
Create `AnalysisPrompt` entity

### PR DESCRIPTION
**Pull request by isxoqovxasanboy**

### CODE REVIEW

**Review by bunyodkamalov — APPROVED** [review]

**Review by rashidovasadbek — APPROVED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Controllers/PromptsController.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    public async ValueTask<IActionResult> DeletePromptById([FromRoute] Guid promptId, CancellationToken cancellationToken = default)
+    {
+        var result = await mediator.Send(
+            new PromptDeleteByIdCommand
+            {
+                PromptId = promptId
+            },
+            cancellationToken
+        );
+        return result ? Ok() : BadRequest();
```
Comment (bunyodkamalov) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Controllers/PromptsController.cs:57]: Add result into Ok() => Ok(result)

**Review by rashidovasadbek — APPROVED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Commands/PromptCreateCommand.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+/// <summary>
+/// Represents a command to create a new organization. 
+/// </summary>
+public record PromptCreateCommand : ICommand<AnalysisPromptDto>
+{
+    /// <summary>
+    /// The data required to create a new organization.
+    /// </summary>
+    public AnalysisPromptDto Prompt { get; set; } = default!;
```
Comment (bunyodkamalov) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Commands/PromptCreateCommand.cs:14]: change Prompt to PromptDto

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Data/SeedDataExtensions.cs [hunk 3]
```diff
@@ -174,7 +177,166 @@ private static async ValueTask SeedDataCustomerFeedbackAsync(AppDbContext appDbC
...
+
+                         ## Customer feedback :
+
+                         {{$customerFeedback}}
+
+                         ## Result
+
+                         """,
+                Version = 4,
+                Revision = 5,
```
Comment (bunyodkamalov) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Data/SeedDataExtensions.cs:336]: Version 5 , Revision - 0

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Data/SeedDataExtensions.cs [hunk 4]
```diff
@@ -174,7 +177,166 @@ private static async ValueTask SeedDataCustomerFeedbackAsync(AppDbContext appDbC
...
+
+                         ## Input
+
+                         {{$customerFeedback}}
+
+                         ## Result
+
+                         """,
+                Version = 3,
+                Revision = 4,
```
Comment (bunyodkamalov) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Data/SeedDataExtensions.cs:295]: Version - 4, Revision - 0

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Data/SeedDataExtensions.cs [hunk 5]
```diff
@@ -174,7 +177,166 @@ private static async ValueTask SeedDataCustomerFeedbackAsync(AppDbContext appDbC
...
+
+                         ## Input
+
+                         {{$customerFeedback}}
+
+                         ## Result
+
+                         """,
+                Version = 2,
+                Revision = 3,
```
Comment (bunyodkamalov) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Data/SeedDataExtensions.cs:269]: Version - 3, Revision - 0

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Data/SeedDataExtensions.cs [hunk 6]
```diff
@@ -174,7 +177,166 @@ private static async ValueTask SeedDataCustomerFeedbackAsync(AppDbContext appDbC
...
+
+                         ## Input
+
+                         {{$customerFeedback}}
+
+                         ## Result
+
+                         """,
+                Version = 1,
+                Revision = 2,
```
Comment (bunyodkamalov) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Data/SeedDataExtensions.cs:242]: Version 2, Revision- 0

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Data/SeedDataExtensions.cs [hunk 7]
```diff
@@ -174,7 +177,166 @@ private static async ValueTask SeedDataCustomerFeedbackAsync(AppDbContext appDbC
...
+
+                         ## Input
+
+                         {{$customerFeedback}}
+
+                         ## Result
+
+                         """,
+                Version = 0,
+                Revision = 1,
```
Comment (bunyodkamalov) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Api/Data/SeedDataExtensions.cs:216]: Version - 1, Revision - 0

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Commands/PromptUpdateCommand.cs [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+/// <summary>
+/// Represents a command to modify an existing organization's information.
+/// </summary>
+public record PromptUpdateCommand : ICommand<AnalysisPromptDto>
+{
+    /// <summary>
+    /// Contains the updated organization data. 
+    /// </summary>
+    public AnalysisPromptDto Prompt { get; set; } = default!;
```
Comment (bunyodkamalov) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Commands/PromptUpdateCommand.cs:14]: change Prompt to PromptDto

**Review by bunyodkamalov — CHANGES_REQUESTED** [review]
Change requested changes

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IPromptRepository.cs [hunk 9]
```diff
@@ -0,0 +1,68 @@
+﻿using System.Linq.Expressions;
+using Feedback.Analyzer.Domain.Common.Commands;
+using Feedback.Analyzer.Domain.Common.Queries;
+using Feedback.Analyzer.Domain.Entities;
+
+namespace Feedback.Analyzer.Persistence.Repositories.Interfaces;
+
+public interface IPromptRepository
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IPromptRepository.cs:8]: Add docs

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Services/IPromptService.cs [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    ValueTask<AnalysisPrompt?> GetByIdAsync(Guid promptId, QueryOptions queryOptions = default, CancellationToken cancellationToken = default);
+
+    ValueTask<AnalysisPrompt> CreateAsync(AnalysisPrompt prompt, CommandOptions commandOptions = default, CancellationToken cancellationToken = default);
+
+    ValueTask<AnalysisPrompt> UpdateAsync(AnalysisPrompt prompt, CommandOptions commandOptions = default, CancellationToken cancellationToken = default);
+
+    ValueTask<AnalysisPrompt?> DeleteAsync(AnalysisPrompt prompt, CommandOptions commandOptions = default, CancellationToken cancellationToken = default);
+
+    ValueTask<AnalysisPrompt?> DeleteByIdAsync(Guid promptId, CommandOptions commandOptions = default, CancellationToken cancellationToken = default);
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Services/IPromptService.cs:24]: Add docs

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Queries/PromptResultGetByCategoryIdQuery.cs [hunk 11]
```diff
@@ -0,0 +1,9 @@
+using Feedback.Analyzer.Application.Common.Prompts.Models;
+using Feedback.Analyzer.Domain.Common.Queries;
+
+namespace Feedback.Analyzer.Application.Common.Prompts.Queries;
+
+public record PromptResultGetByCategoryIdQuery : IQuery<ICollection<PromptResultDto>>
+{
+    public Guid CategoryId { get; set; }
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Queries/PromptResultGetByCategoryIdQuery.cs:9]: Add docs

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Models/PromptResultDto.cs [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+    public int Version { get; set; }
+    
+    public int Revision { get; set; }
+    
+    public int ExecutionsCount { get; set; }
+    
+    public TimeSpan AverageExecutionTime { get; set; }
+    
+    public int AverageAccuracy { get; set; }
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Models/PromptResultDto.cs:16]: Add docs

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Models/AnalysisPromptDto.cs [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+    public int Version { get; set; }
+    
+    public int Revision { get; set; }
+    
+    public Guid CategoryId { get; set; }
+    
+    public DateTimeOffset CreatedTime { get; set; }
+    
+    public DateTimeOffset ModifiedTime { get; set; }
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Models/AnalysisPromptDto.cs:22]: Add docs

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by bunyodkamalov — APPROVED** [review]

**Review by sulton-max — APPROVED** [review]

**Review by Ilxom-22 — APPROVED** [review]

**Pull request merged**